components/helpers.js

```diff
@@ -1,37 +1,34 @@
 const EResult = require('../resources/EResult.js');
 
 exports.isSteamID = function(input) {
-	var keys = Object.keys(input);
+	let keys = Object.keys(input);
 	if (keys.length != 4) {
 		return false;
 	}
 
 	// Make sure it has the keys we expect
-	keys = keys.filter(function(item) {
-		return ['universe', 'type', 'instance', 'accountid'].indexOf(item) != -1;
-	});
-
-	return keys.length == 4;
+	keys.sort();
+	return keys.join(',') == 'accountid,instance,type,universe';
```

Comment:
```suggestion
	for (let key of keys) {
		switch(key) {
			case 'accountid':
			case 'instance':
			case 'type':
			case 'universe':
				continue;
			default:
				return false;
		}
	}
return true;
```
![image](https://user-images.githubusercontent.com/4411977/127515391-960f0613-6080-4afe-b3f5-4bc97e6eaa5b.png)

This works, because we already check the keys length and object keys must be distinct.

Reply:
what about this? : `return ['universe', 'type', 'instance', 'accountid'].every(prop=> input[prop] != undefined)`
I think we should ignore custom properties

Reply:
That should also work